packages/stk/stk_balance/stk_balance/balance.hpp

```diff
@@ -42,12 +42,27 @@ namespace stk { namespace mesh { class BulkData; } }
 #include <string>
 #include "mpi.h"
 #include <stk_balance/internal/balanceDefaults.hpp>
+#include "balanceUtils.hpp"               // for BalanceSettings, etc
 
 namespace stk
 {
 namespace balance
 {
 
+class StkBalanceSettings : public GraphCreationSettings
+{
+public:
+    StkBalanceSettings()
```

Comment:
It suffices, and is better, to say `StkBalanceSettings() = default;`.  It suffices because defaulted or implicitly defined default constructors [automatically call their base class default constructors](https://en.cppreference.com/w/cpp/language/default_constructor).  It's better for two reasons:

1. It declares intent: "My class has no invariants other than the base class' invariants."  (Defining an explicit constructor marks the class as having invariants.)
2. If a class would otherwise be trivially default constructible, defining the constructor explicitly without `= default` would make the class not trivially default constructible.

You want to collect those "trivially `$X` constructible" attributes because they can help with compiler optimization.  It's probably less useful for this class because it looks like it has virtual methods, but it's a good habit (esp. trivially copy constructible, because that more or less says the compiler can copy the bits directly).